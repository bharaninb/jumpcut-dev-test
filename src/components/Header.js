import AppBar from '@material-ui/core/AppBar';
import Toolbar from '@material-ui/core/Toolbar';
import Typography from '@material-ui/core/Typography';
import PropTypes from 'prop-types';
import React from 'react';

export const HeaderComponent = ({ title }) => (
  <AppBar position="absolute">
    <Toolbar>
      <Typography component="h1" variant="h6" color="inherit" noWrap>
        {title}
      </Typography>
    </Toolbar>
  </AppBar>
);

HeaderComponent.propTypes = {
  title: PropTypes.string.isRequired,
};
