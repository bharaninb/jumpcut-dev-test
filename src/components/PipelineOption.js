import FormControl from '@material-ui/core/FormControl';
import FormControlLabel from '@material-ui/core/FormControlLabel';
import FormLabel from '@material-ui/core/FormLabel';
import Radio from '@material-ui/core/Radio';
import RadioGroup from '@material-ui/core/RadioGroup';
import { withStyles } from '@material-ui/core/styles';
import PropTypes from 'prop-types';
import React from 'react';

const styles = theme => ({
  root: {
    display: 'flex',
  },
  formControl: {
    margin: theme.spacing.unit * 3,
  },
  group: {
    margin: `${theme.spacing.unit}px 0`,
  },
});

class PipelineOption extends React.Component {
  handleChange = (event) => {
    const { onChange } = this.props;
    onChange(event.target.value);
  };

  render() {
    const { classes, pipelineOption } = this.props;

    return (
      <div className={classes.root}>
        <FormControl component="fieldset" className={classes.formControl}>
          <FormLabel component="legend">PipelineOption</FormLabel>
          <RadioGroup
            aria-label="PipelineOption"
            name="PipelineOption"
            className={classes.group}
            value={pipelineOption}
            onChange={this.handleChange}
          >
            <FormControlLabel value="none" control={<Radio />} label="None" />
            <FormControlLabel value="accumulator" control={<Radio />} label="Accumulator" />
            <FormControlLabel value="isEven" control={<Radio />} label="IsEven" />
          </RadioGroup>
        </FormControl>
      </div>
    );
  }
}

PipelineOption.propTypes = {
  classes: PropTypes.shape().isRequired,
  pipelineOption: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
};

export const PipelineOptionComponent = withStyles(styles)(PipelineOption);
