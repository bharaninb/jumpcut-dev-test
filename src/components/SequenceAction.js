import Button from '@material-ui/core/Button';
import { withStyles } from '@material-ui/core/styles';
import PropTypes from 'prop-types';
import React from 'react';

const styles = theme => ({
  root: {
    padding: theme.spacing.unit * 3,
    textAlign: 'center',
  },
});

class SequenceAction extends React.Component {
  handleNext = () => {
    const { handleNext, appSequence } = this.props;
    handleNext(appSequence.sequenceGen);
  };

  handleActive = () => {
    const { activateSequence, appSequence } = this.props;
    activateSequence(appSequence.selectedSequence, appSequence.pipelineOption);
  };

  render() {
    const { classes, appSequence } = this.props;

    if (appSequence.isActive) {
      return (
        <div className={classes.root}>
          <Button variant="contained" color="primary" size="large" onClick={this.handleNext} id="nextBtn">
              Next
          </Button>
        </div>
      );
    }

    return (
      <div className={classes.root}>
        <Button variant="contained" color="primary" size="large" onClick={this.handleActive} id="activateBtn">
            Activate
        </Button>
      </div>
    );
  }
}

SequenceAction.propTypes = {
  classes: PropTypes.shape().isRequired,
  appSequence: PropTypes.shape().isRequired,
  activateSequence: PropTypes.func.isRequired,
  handleNext: PropTypes.func.isRequired,
};

export const SequenceActionComponent = withStyles(styles)(SequenceAction);
