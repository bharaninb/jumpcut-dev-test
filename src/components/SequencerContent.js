import Button from '@material-ui/core/Button';
import Grid from '@material-ui/core/Grid';
import Paper from '@material-ui/core/Paper';
import { withStyles } from '@material-ui/core/styles';
import PropTypes from 'prop-types';
import React from 'react';
import { uid } from 'react-uid';

const styles = theme => ({
  root: {
    flexGrow: 1,
    padding: theme.spacing.unit,
  },
  paper: {
    padding: theme.spacing.unit * 2,
    textAlign: 'center',
    color: theme.palette.text.secondary,
  },
  button: {
    textAlign: 'center',
  },
});

class SequencerContent extends React.Component {
  handleClick = () => {
    const { handleNext, sequenceGen } = this.props;
    handleNext(sequenceGen);
  };

  render() {
    const { classes, sequenceValues } = this.props;

    const valueItems = sequenceValues.map((value, index) => (
      <Grid item key={uid(value, index)}>
        <Paper className={classes.paper}>{value}</Paper>
      </Grid>
    ));

    return (
      <div className={classes.root}>
        <Grid container spacing={24}>
          {valueItems}
          <Grid item xs={12} className={classes.button}>
            <Button variant="contained" color="primary" size="large" onClick={this.handleClick} id="nextBtn">
              Next
            </Button>
          </Grid>
        </Grid>
      </div>
    );
  }
}

SequencerContent.propTypes = {
  classes: PropTypes.shape().isRequired,
  sequenceValues: PropTypes.arrayOf(PropTypes.number).isRequired,
  handleNext: PropTypes.func.isRequired,
  sequenceGen: PropTypes.shape().isRequired,
};

export const SequencerContentComponent = withStyles(styles)(SequencerContent);
