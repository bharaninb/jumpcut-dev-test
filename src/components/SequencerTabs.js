import AppBar from '@material-ui/core/AppBar';
import { withStyles } from '@material-ui/core/styles';
import Tab from '@material-ui/core/Tab';
import Tabs from '@material-ui/core/Tabs';
import PropTypes from 'prop-types';
import React from 'react';

import { availableSequencers } from '../constants';

const styles = theme => ({
  root: {
    flexGrow: 1,
    width: '100%',
    backgroundColor: theme.palette.background.paper,
    marginTop: '65px',
  },
});

class SequencerTabs extends React.Component {
  handleChange = (event, value) => {
    const { onChange } = this.props;
    onChange(value);
  };

  render() {
    const { classes, selectedSequence } = this.props;

    const tabList = availableSequencers.map(sequence => (
      <Tab label={sequence.name} value={sequence.id} key={sequence.id} />
    ));

    return (
      <div className={classes.root}>
        <AppBar position="static" color="default">
          <Tabs
            id="sequenceTab"
            value={selectedSequence}
            onChange={this.handleChange}
            indicatorColor="primary"
            textColor="primary"
            scrollable
            scrollButtons="auto"
          >
            {tabList}
          </Tabs>
        </AppBar>
      </div>
    );
  }
}

SequencerTabs.propTypes = {
  classes: PropTypes.shape().isRequired,
  selectedSequence: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
};

export const SequencerTabsComponent = withStyles(styles)(SequencerTabs);
