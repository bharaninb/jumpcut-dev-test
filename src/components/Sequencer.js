import PropTypes from 'prop-types';
import React from 'react';

import { SequencerContentComponent } from './SequencerContent';
import { SequencerTabsComponent } from './SequencerTabs';

class SequencerComponent extends React.Component {
  componentDidMount() {
    const { sequenceGen, selectSequence, selectedSequence } = this.props;
    const isEmptyGen = Object.keys(sequenceGen).length === 0;
    if (isEmptyGen) {
      selectSequence(selectedSequence);
    }
  }

  render() {
    const {
      selectedSequence,
      sequenceValues,
      selectSequence,
      sequenceGen,
      fetchNextSequenceValue,
    } = this.props;

    return (
      <main>
        <SequencerTabsComponent selectedSequence={selectedSequence} onChange={selectSequence} />
        <SequencerContentComponent
          sequenceValues={sequenceValues}
          sequenceGen={sequenceGen}
          handleNext={fetchNextSequenceValue}
        />
      </main>
    );
  }
}

SequencerComponent.propTypes = {
  selectedSequence: PropTypes.string.isRequired,
  sequenceValues: PropTypes.arrayOf(PropTypes.number).isRequired,
  selectSequence: PropTypes.func.isRequired,
  sequenceGen: PropTypes.shape().isRequired,
  fetchNextSequenceValue: PropTypes.func.isRequired,
};

export { SequencerComponent };
