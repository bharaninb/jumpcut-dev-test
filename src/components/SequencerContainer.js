import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';

import { selectSequence, fetchNextSequenceValue } from '../redux/actions';
import { SequencerComponent } from './Sequencer';

const mapStateToProps = (state) => {
  const { selectedSequence, sequenceValues, sequenceGen } = state.appSequence;
  return {
    selectedSequence,
    sequenceValues,
    sequenceGen,
  };
};

const mapDispatchToProps = dispatch => ({
  selectSequence: bindActionCreators(selectSequence, dispatch),
  fetchNextSequenceValue: bindActionCreators(fetchNextSequenceValue, dispatch),
});

export const SequencerContainer = connect(
  mapStateToProps,
  mapDispatchToProps,
)(SequencerComponent);
