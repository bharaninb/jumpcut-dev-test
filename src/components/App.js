import React from 'react';

import { HeaderComponent } from './Header';
import { SequencerContainer } from './SequencerContainer';

export const AppComponent = () => (
  <React.Fragment>
    <HeaderComponent title="Jumpcut Dev Test" />
    <SequencerContainer />
  </React.Fragment>
);
