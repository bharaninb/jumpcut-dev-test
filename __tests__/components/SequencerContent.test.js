import { shallow, mount } from 'enzyme';
import React from 'react';

import { SequencerContentComponent } from '../../src/components/SequencerContent';

describe('<SequencerContentComponent />', () => {
  it('should call handleNext with sequenceGen on "next" click', () => {
    const handleNext = jest.fn();
    const sequenceGen = { test: 'test' };
    const component = mount(
      <SequencerContentComponent
        sequenceValues={[2]}
        sequenceGen={sequenceGen}
        handleNext={handleNext}
      />,
    );
    component.find('button#nextBtn').simulate('click');

    expect(handleNext).toBeCalledWith(sequenceGen);
  });

  it('should match snapshot', () => {
    const handleNext = jest.fn();
    const component = shallow(
      <SequencerContentComponent sequenceValues={[2]} sequenceGen={{}} handleNext={handleNext} />,
    );
    expect(component).toMatchSnapshot();
  });
});
