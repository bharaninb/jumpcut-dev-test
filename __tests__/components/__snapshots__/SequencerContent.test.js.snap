// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SequencerContentComponent /> should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WithStyles(SequencerContent)
    handleNext={[MockFunction]}
    sequenceGen={Object {}}
    sequenceValues={
      Array [
        2,
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "classes": Object {
        "button": "SequencerContent-button-3",
        "paper": "SequencerContent-paper-2",
        "root": "SequencerContent-root-1",
      },
      "handleNext": [MockFunction],
      "sequenceGen": Object {},
      "sequenceValues": Array [
        2,
      ],
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "classes": Object {
          "button": "SequencerContent-button-3",
          "paper": "SequencerContent-paper-2",
          "root": "SequencerContent-root-1",
        },
        "handleNext": [MockFunction],
        "sequenceGen": Object {},
        "sequenceValues": Array [
          2,
        ],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
